### DISCUSSION TITLE
新版本的 Hook 机制是否有更改？

### DISCUSSION THREAD

**Discussion by Robotxm** [LSPosed/LSPosed#2607] [Q&A] [answered]
发现编写的模块在 1.8.6 (6712) 上运行良好，但在更新的版本上不能正常工作。尝试打了一些日志发现和 Retrofit 以及 Kotlin 协程有关。

我的模块使用了协程，因此 Retrofit 是最新的原生支持协程的版本。新版本上的表现看起来会使用宿主 app 的 Retrofit 类，而宿主 app 使用的 Retrofit 版本似乎比较低，出现了 `Unable to create call adapter for class java.lang.Object` 的错误。

因此想问一下对于宿主和模块中引用的相同的类，新版本 LSPosed 是否改为加载宿主的？

**Comment by yujincheng08** [comment, marked as answer]
不会。宿主的类加载器和模块的类加载器是相互隔离的。

**Reply by Robotxm** [reply]
有点怪，我现在升 LSPosed 之后模块 Hook 部分用 Retrofit 找不到协程的 adapter，只能先暂时用 Retrofit 自己的 `Call` 凑合一下
